### DISCUSSION TITLE
Installation in Docker (Synology) via Tasks

### DISCUSSION THREAD

**Discussion by update-freak** [dec0dOS/zero-ui#86] [Q&A] [answered]
Hi,
in Synology it is possible to install Docker containers via Task. (Here is an example: https://mariushosting.com/how-to-install-adguard-on-your-synology-nas/)
I'm not sure what is the correct code for this docker container. My proposal is here:

`docker run -d --name=ZeroUI \
-v /volume1/docker/zero-ui/controller_data:/var/lib/zerotier-one \
-v /volume1/docker/zero-ui/zero-ui_data:/app/backend/data \
-- ZU_CONTROLLER_ENDPOINT=http://DYNDNS-ADRESSE:9993/
-- ZU_SECURE_HEADERS=true
-- ZU_DEFAULT_USERNAME=felix
-- ZU_DEFAULT_PASSWORD=password
--restart always \
dec0dos/zero-ui`

Is this correct? If not what will be the right code?

**Comment by dec0dOS** [comment, marked as answer]
Hello, @update-freak.
You should use the docker-compose on Synology 
https://linuxhint.com/docker-compose-with-synology/
https://www.wundertech.net/how-to-update-docker-compose-on-a-synology-nas/

**Comment by update-freak** [comment]
ok thanks. I installed Portainer to use the docker-compose.
Synology can set up reverse proxies in the settings, so caddy:latest is not necessary or?

**Reply by dec0dOS** [reply]
No, it's not. Check the docs (README.md) on how to disable the HTTPS and caddy.

**Comment by update-freak** [comment]
All right. So this is correct:?
ports:
      - "4000"

Or
ports:
      - "4000:4000"

Or
ports:
      -"4000:4000/tcp"

**Comment by update-freak** [comment]
![portainer error](https://user-images.githubusercontent.com/40196995/174436289-ceb722fb-b033-43eb-86b6-7f3ba62d7e82.png)
I got this error:
